tests/utils/httpd.py

```diff
@@ -35,6 +36,35 @@ class ContentMD5Handler(TestRequestHandler):
     checksum_header = "Content-MD5"
 
 
+class PushRequestHandler(SimpleHTTPRequestHandler):
+    def _chunks(self):
+        while True:
+            data = self.rfile.readline(65537)
+            chunk_size = int(data[:-2], 16)
+            if chunk_size == 0:
+                return
+            data = self.rfile.read(chunk_size)
+            yield data
+            self.rfile.read(2)
+
+    def do_POST(self):
+        chunked = self.headers.get("Transfer-Encoding", "") == "chunked"
+        path = self.translate_path(self.path)
+        try:
+            os.makedirs(os.path.dirname(path), exist_ok=True)
+            with open(path, "wb") as fd:
+                if chunked:
+                    for chunk in self._chunks():
+                        fd.write(chunk)
+                else:
+                    size = int(self.headers.get("Content-Length", 0))
+                    fd.write(self.rfile.read(size))
+        except (IOError, OSError) as e:
```

Comment:
*Nitpicky*:
We don't need to support Python2.7, so, I don't think we need both here as `IOError` is an alias for `OSError` on Python3.
```suggestion
        except OSError as e:
```